packages/serve/test/auth/cannerPATAuthenticator.spec.ts

```diff
@@ -0,0 +1,237 @@
+import * as sinon from 'ts-sinon';
+import { IncomingHttpHeaders } from 'http';
+import { Request, BaseResponse } from 'koa';
+import { CannerPATAuthenticator } from '@vulcan-sql/serve/auth';
+import { AuthResult, AuthStatus, KoaContext } from '@vulcan-sql/serve/models';
+
+const wrappedAuthCredential = async (
+  ctx: KoaContext,
+  options: any,
+  resolveValue: any = null
+): Promise<AuthResult> => {
+  const authenticator = new CannerPATAuthenticator({ options }, '');
+  if (resolveValue) {
+    sinon.default
+      .stub(authenticator, <any>'fetchCannerUser')
+      .resolves(resolveValue);
+  }
+  await authenticator.activate();
+  return await authenticator.authCredential(ctx);
+};
+
+const getTokenInfo = async (
+  ctx: KoaContext,
+  options: any
+): Promise<Record<string, any>> => {
+  const authenticator = new CannerPATAuthenticator({ options }, '');
+  await authenticator.activate();
+  return await authenticator.getTokenInfo(ctx);
+};
+
+const expectIncorrect = {
+  status: AuthStatus.INDETERMINATE,
+  type: 'canner-pat',
+};
+const expectFailed = (
+  message = 'authenticate user by "canner-pat" type failed.'
+) => ({
+  status: AuthStatus.FAIL,
+  type: 'canner-pat',
+  message,
+});
+const mockOptions = {
+  'canner-pat': {
+    host: 'mockHost',
+    port: 3000,
+    ssl: false,
+  },
+};
+const invalidToken = Buffer.from('clientId:clientSecret').toString('base64');
```

Comment:
great way to create the invalid token and also make members know the token content pattern.